python/python_helper_sympy_test.py

```diff
@@ -91,13 +95,29 @@ def test_string_conversion_no_complex(
     def test_json_conversion(self, a_sub: str) -> None:
         sympy_expr = phs.convert_string_to_sympy(a_sub, ["n", "m"], allow_complex=True)
         # Check that json serialization works
-        json_expr = phs.sympy_to_json(sympy_expr)
-        assert type(json.dumps(json_expr)) == str
+        json_expr = json.dumps(phs.sympy_to_json(sympy_expr), allow_nan=False)
```

Comment:
(drive by comment, ignore if not helpful)

`json.dumps(sympy_to_json(...))` seems a bit like "boil the boiling water" to me.  Is this b/c of no-NaN-JSON-thing?  (e.g., https://stackoverflow.com/questions/6601812/sending-nan-in-json)

Maybe a comment here would clarify?

Reply:
So the dumps turns the `sympy_to_json(...)` into a serialized string, simulating what PL does on the backend for a question. `sympy_to_json` just turns a sympy expression into a format that can be JSON serialized, not doing the serialization itself. This part of the test is just asserting that operation all worked as expected.

Reply:
Got it!  (`sympy_prep_for_json` perhaps a more accurate name then---definitely not suggesting you change it!)

Reply:
or maybe `sympy_to_pyobj`? its just converting it to a python object/dict or something like that, right?

Reply:
`sympy_to_pyobj` is more accurate, but I'm against changing the name of this for backwards compatibility and consistency with the `to_json` and `from_json` functions.